transports/email/src/main/java/org/mule/transport/email/RetrieveMessageReceiver.java

```diff
@@ -166,11 +162,50 @@ protected void doStop() throws MuleException
     @Override
     protected void doStart() throws MuleException
     {
-        super.doStart();
-        synchronized (folderLock)
+        try
         {
-            folder.addMessageCountListener(this);
+            getConnector().getRetryPolicyTemplate().execute(new RetryCallback()
+            {
+                @Override
+                public void doWork(RetryContext retryContext) throws MuleException
+                {
+                    synchronized (connectionLock)
+                    {
+                        if (RetrieveMessageReceiver.this.connectionEstablished)
+                        {
+                            RetrieveMessageReceiver.super.doStart();
+                            synchronized (folderLock)
+                            {
+                                folder.addMessageCountListener(RetrieveMessageReceiver.this);
+                            }
+                        }
+                        else
+                        {
+                            throw new NotYetConnectedException();
+                        }
+                    }
+                }
+
+                @Override
+                public String getWorkDescription() {
+                    return "Trying to start the retrieve message retriever";
+                }
+
+                @Override
+                public Object getWorkOwner() {
+                    return RetrieveMessageReceiver.this;
+                }
+            }, getWorkManager());
+        }
+        catch (MuleException me)
+        {
+            throw me;
         }
+        catch (Exception e)
+        {
+            // Ignore other exceptions
```

Comment:
Why?

Reply:
Now all caught exceptions are wrapped in a LifecycleException